pkgs/os-specific/linux/kernel/generic.nix

```diff
@@ -90,7 +93,9 @@ let
 
     depsBuildBuild = [ buildPackages.stdenv.cc ];
     nativeBuildInputs = [ perl ]
-      ++ lib.optionals (stdenv.lib.versionAtLeast version "4.16") [ bison flex ];
+      ++ lib.optionals (stdenv.lib.versionAtLeast version "4.16") [ bison flex ]
+      # For GCC plugins support.
+      ++ lib.optionals (stdenv.lib.versionAtLeast version "4.18") [ gmp libmpc mpfr ];
```

Comment:
These deps are added via manual-config.nix since a04d8532c2385ab3e217d06bd3592d0386b6caa3 Is that no longer sufficient? In any case, it'd be nice to add them in only one place, whichever is most appropriate. I'm also unsure why this would need to be version guarded? gcc plugins should be supported for 4.8 onwards, unless something has changed in this regard.

Reply:
My understanding is that generic.nix builds its .config without using the facilities from manual-config.nix (it overrides configfile). As such I'm not sure whether we can DRY this :/

Agreed about the version guarding, I just guarded it to match the removed condition in hardened-config.nix. We could probably remove it -- the deps are in stdenv anyway if I'm not mistaken, they're not really going to hurt much :) I'll fix that tonight.

Reply:
Alright, I guess something has changed in the way things are built, then; I haven't really kept up to date with the generic kernel builder. Not a huge deal, but a little unfortunate.